src/compiler/transformers/es2015.ts

```diff
@@ -1571,6 +1578,51 @@ namespace ts {
                         break;
                 }
             }
+
+            /**
+             * Lazily creates the way class members will access the class prototype.
+             */
+            function getPrototypeStorageName() {
+                if (prototypeStorageName) {
+                    return prototypeStorageName;
+                }
+
+                const originalPrototypeAccess = createPropertyAccess(getInternalName(node), "prototype");
+
+                // If the class has exactly one non-static member, it'll access prototype members on itself:
+                //   ClassName.prototype.member = ...
+                if (containsExactlyOne(node.members, classMemberAssignsToPrototype)) {

```

Comment:
Aha, going through the baselines, this has a bug!

```ts
class Ha {
    get x() { return ''; }
    set x(value: string) {}
}
```

This logic should check for how many unique names are in the node members.